packages/web-components/src/components/va-combo-box/va-combo-box.tsx

```diff
@@ -0,0 +1,251 @@
+
+/**
+ * @componentName Combo Box
+ * @maturityCategory dont_use
+ * @maturityLevel proposed
+ * @guidanceHref form/combo-box
+ * @translations English
+ * @translations Spanish
+ */
+
+@Component({
+  tag: 'va-combo-box',
+  styleUrl: 'va-combo-box.scss',
+  shadow: true,
+})
+export class VaComboBox {
+  @Element() el: any;
+
+  @State() options: Array<Node>;
+
+  @State() labelNode: HTMLLabelElement;
+
+  /**
+   * Whether or not this is a required field.
+   */
+  @Prop() required?: boolean = false;
+
+  /**
+   * The combo box component will be disabled / read-only.
+   */
+  @Prop() disabled?: boolean = false;
+
+  /**
+   * Text label for the field.
+   */
+  @Prop() label!: string;
+
+  /**
+   * The placeholder string.
+   */
+  @Prop() placeholder?: string;
+
+  /**
+   * Name attribute for the select field.
+   */
+  @Prop() name!: string;
+
+  /**
+   * Selected value (will get updated on select).
+   */
+  @Prop({ reflect: true, mutable: true }) value?: string;
+
+  /**
+   * Error message to display. When defined, this indicates an error.
+   */
+  @Prop() error?: string;
+
+  /**
+   * Whether or not `aria-invalid` will be set on the inner select. Useful when
+   * composing the component into something larger, like a date component.
+   */
+  @Prop() invalid?: boolean = false;
+
+  /**
+   * Whether or not to fire the analytics events
+   */
+  @Prop() enableAnalytics?: boolean = false;
+
+  /**
+   * Optional hint text.
+   */
+  @Prop() hint?: string;
+
+  /**
+   * An optional message that will be read by screen readers when the select is focused.
+   */
+  @Prop() messageAriaDescribedby?: string;
+
+  /**
+   * The event emitted when the selected value changes
+   */
+  @Event() vaSelect: EventEmitter;
+
+  /**
+   * Whether an error message should be shown - set to false when this component is used inside va-date or va-memorable-date
+   * in which the error for the va-select will be rendered outside of va-select
+   */
+  @Prop() showError?: boolean = true;
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * option is selected and enableAnalytics is true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  connectedCallback() {
+    i18next.on('languageChanged', () => {
+      forceUpdate(this.el);
+    });
+    if (!this.label) {
+      throw new Error('va-combo-box: label is a required property');
+    }
+  }
+
+  componentDidLoad() {
+    const comboBoxElement = this.el.shadowRoot.querySelector('.usa-combo-box');
+    const labelElement = this.el.shadowRoot.querySelector('label');
+    if (comboBoxElement) {
+      comboBox.init(comboBoxElement, labelElement);
+      comboBox.on(comboBoxElement);
+    }
+    const inputElement = this.el.shadowRoot.querySelector('input');
+    if (inputElement && this.error) {
+      inputElement.classList.add('usa-input--error');
```

Comment:
Do you mind explaining a little more why displaying the error needs to be done in a component lifecycle instead of just using conditional markup in the render?

Reply:
that is is due to `<input>` being added by the `init` function. Have to wait until `init` is called, `input` rendered and then assign `error` class if that prop is present